go/vt/discovery/healthcheck_test.go

```diff
@@ -63,6 +63,77 @@ func init() {
 	refreshInterval = time.Minute
 }
 
+func TestNewVTGateHealthCheckFilters(t *testing.T) {
+	defer func() {
+		KeyspacesToWatch = nil
+		tabletFilters = nil
+		tabletFilterTags = nil
+	}()
+
+	testCases := []struct {
+		name                string
+		keyspacesToWatch    []string
+		tabletFilters       []string
+		tabletFilterTags    map[string]string
+		expectedError       string
+		expectedFilterTypes []any
+	}{
+		{
+			name: "noFilters",
+		},
+		{
+			name:                "tabletFilters",
+			tabletFilters:       []string{"ks1|-80"},
+			expectedFilterTypes: []any{&FilterByShard{}},
```

Comment:
How does this confirm any specific behavior? i.e. that we filter out any tablets in that `ks1/-80` shard?

Reply:
@mattlord this test confirms the right filter is provided. The individual filter implementations are tested here:
1. [`FilterByShard`](https://github.com/vitessio/vitess/blob/main/go/vt/discovery/topology_watcher_test.go#L301-L371)
2. [`FilterByKeyspace`](https://github.com/vitessio/vitess/blob/main/go/vt/discovery/topology_watcher_test.go#L392-L463)
3. [`FilterByTags`](https://github.com/vitessio/vitess/blob/main/go/vt/discovery/topology_watcher_test.go#L581-L606)

Reply:
Thanks! I must be blind 🤣